File: Backend/MicroServices/Investor-Matching-and-Presentation/src/main/java/com/example/Investor/Matching/and/Presentation/service/MatchingService.java
package com.example.Investor.Matching.and.Presentation.service;

import com.example.Investor.Matching.and.Presentation.dto.MatchResponse;

import java.util.List;

public interface MatchingService {

    void runMatching(Long startupExecutionId, Long startupUserId);

    void runMatchingForNewInvestor(Long investorExecutionId, Long investorUserId);

    void runMatchingForAll();

    List<MatchResponse> getMatchesForInvestor(Long investorUserId);

    List<MatchResponse> getMatchesForStartup(Long startupUserId);

    List<MatchResponse> getAllMatches();
}

File: frontend/src/services/matchingService.ts
import axios from "axios";

const matchingServiceApi = axios.create({
  baseURL: "https://investor-matching-and-presentation.onrender.com",
  timeout: 30000,
  headers: { "Content-Type": "application/json" },
});

matchingServiceApi.interceptors.request.use((config) => {
  if (typeof window !== "undefined") {
    const token = localStorage.getItem("token");
    if (token) config.headers.Authorization = `Bearer ${token}`;
  }
  return config;
});

export const matchingService = {
  getMatchesForInvestor: (investorUserId: number) =>
    matchingServiceApi.get(`/api/matching/investor/${investorUserId}`),

  getMatchesForStartup: (startupUserId: number) =>
    matchingServiceApi.get(`/api/matching/startup/${startupUserId}`),

  getAllMatches: () =>
    matchingServiceApi.get("/api/matching/admin/all"),
};